kubernetes/samples/scripts/domain-lifecycle/helper.sh

```diff
@@ -0,0 +1,237 @@
+# !/bin/sh
+# Copyright (c) 2020, Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+#
+
+#
+# Function to create server start policy patch string
+# $1 - Domain resource in json format
+# $2 - Name of server whose policy will be patched
+# $3 - Policy value 
+# $4 - Return value containing server start policy patch string
+#
+function createServerStartPolicyPatch {
+  local domainJson=$1
+  local serverName=$2
+  local policy=$3
+  local __result=$4
+  local __currentPolicy=$5
+  local currentServerStartPolicy=""
+
+  # Get server start policy for this server
+  managedServers=$(echo ${domainJson} | jq -cr '(.spec.managedServers)')
+  if [ "${managedServers}" != "null" ]; then
+    extractPolicyCmd="(.spec.managedServers[] \
+      | select (.serverName == \"${serverName}\") | .serverStartPolicy)"
+    currentServerStartPolicy=$(echo ${domainJson} | jq "${extractPolicyCmd}")
+  fi
+  if [ -z "${currentServerStartPolicy}" ]; then
+    # Server start policy doesn't exist, add a new policy
+    addPolicyCmd=".[.| length] |= . + {\"serverName\":\"${serverName}\", \
+      \"serverStartPolicy\":\"${policy}\"}"
+    serverStartPolicyPatch=$(echo ${domainJson} | jq .spec.managedServers | jq -c "${addPolicyCmd}")
+  else
+    # Server start policy exists, replace policy value 
+    replacePolicyCmd="(.spec.managedServers[] \
+      | select (.serverName == \"${serverName}\") | .serverStartPolicy) |= \"${policy}\""
+    servers="(.spec.managedServers)"
+    serverStartPolicyPatch=$(echo ${domainJson} | jq "${replacePolicyCmd}" | jq -cr "${servers}")
+  fi
+  eval $__result="'${serverStartPolicyPatch}'"
+  eval $__currentPolicy=${currentServerStartPolicy}
+}
+
+#
+# Function to create patch string for updating replica count
+# $1 - Domain resource in json format
+# $2 - Name of cluster whose replica count will be patched
+# $3 - operatation string indicating whether to increment or decrement count
+# $4 - Return value containing replica update patch string
+# $5 - Retrun value containing updated replica count
+#
+function createReplicaPatch {
+  local domainJson=$1
+  local clusterName=$2
+  local operation=$3
+  local __result=$4
+  local __relicaCount=$5
```

Comment:
Minor comment: mispelling. relica --> replica

Reply:
Thanks for catching this. Fixed.